src/lmnt/api.py

```diff
@@ -95,12 +123,79 @@ async def synthesize(self, text, voice, **kwargs):
     form_data = aiohttp.FormData()
     form_data.add_field('text', text)
     form_data.add_field('voice', voice)
-    form_data.add_field('seed', kwargs.pop('seed', 0))
-    form_data.add_field('format', kwargs.pop('format', 'wav'))
-    form_data.add_field('speed', kwargs.pop('speed', 1.0))
+    form_data.add_field('seed', kwargs.get('seed', 0))
+    form_data.add_field('format', kwargs.get('format', 'wav'))
+    form_data.add_field('speed', kwargs.get('speed', 1.0))
+
+    is_multipart_response = False
+    extras = ''
+
+    has_duration = kwargs.get('durations', False)
+    if has_duration:
+      extras = 'alignment'
+      is_multipart_response = True
+
+    if extras:
+      form_data.add_field('extras', extras)
+
     async with self._session.post(url, data=form_data, headers=self._build_headers()) as resp:
-      await self._handle_response_errors(resp)
-      return await resp.read()
+      if is_multipart_response:
+        response = await self._parse_multipart_alignment_response(resp)
+        word_durations = self._transform_to_word_durations(response['duration'], response['phonemes'])
+        return {
+          'durations': word_durations,
+          'audio': response['audio']
+        }
+
+      else:
+        await self._handle_response_errors(resp)
+        return await resp.read()
+
+  def _transform_to_word_durations(self, durations, phonemes):
+    if not durations or not phonemes or len(durations) != len(phonemes):
+      raise ValueError("Invalid word transformation input data.")
+
+    result = []
+    temp_phonemes, temp_durations = [], []
+    total_duration, start = 0, 0
+
+    for index, (dur, phon) in enumerate(zip(durations, phonemes)):
+      temp_phonemes.append(phon)
+      duration_samples = dur * _SAMPLES_PER_FRAME
+      temp_durations.append(duration_samples)
+      total_duration += duration_samples
+
+      # Break words at any whitespace phoneme.
+      if phon == " " or ((index < len(durations) - 1) and phonemes[index + 1] == " "):
```

Comment:
I think we can eliminate the peek and simplify this if we move before line 163?

e.g.

for all the phones:
  - if current is whitespace:
    - emit current accumulation if anything accumulated
     - emit current current whitespace token
  - else:
    - accumulate

and then after the for, make sure to emit any remaining accumulation (to catch case if last is not whitespace)

Reply:
Done.